ReactSkia/components/RSkComponentText.cpp

```diff
@@ -51,84 +52,135 @@ inline int getNumberOfLines(std::vector<LineMetrics>& metrics, int maxNumberOfLi
         }
     }
-    if ((minNumberOfLines > 0) && (minNumberOfLines < maxNumberOfLines))
+    if ((minNumberOfLines) && (maxNumberOfLines) && (minNumberOfLines < maxNumberOfLines))
         return minNumberOfLines;
-    
+        
     return maxNumberOfLines;
 }
 
+inline int getLines(std::vector<LineMetrics>& metrics, Float frameHeight) {
+    int i=0;
+    int minNumberOfLines = 0;
+    for(; i< metrics.size(); i++) {
+        if ((metrics[i]).fBaseline > frameHeight) {
+            minNumberOfLines = (metrics[i]).fLineNumber;
+            break;
+        }
+    }
+    return minNumberOfLines;
+}
+ 
+inline SkPaint convertTextColor(SharedColor textColor) {
+    SkPaint paint;
+    float ratio = 255.9999;
+    auto color = colorComponentsFromColor(textColor);
+    paint.setColor(SkColorSetARGB( color.alpha * ratio,
+                                        color.red * ratio,
+                                        color.green * ratio,
+                                        color.blue * ratio));
+    paint.setAntiAlias(true);
+    return paint;
+}
+
 void RSkComponentParagraph::OnPaint(SkCanvas *canvas) {
+    SkAutoCanvasRestore(canvas, true);
     auto component = getComponentData();
     auto state = std::static_pointer_cast<ParagraphShadowNode::ConcreteStateT const>(component.state);
     auto const &props = *std::static_pointer_cast<ParagraphProps const>(component.props);
     auto data = state->getData();
+    auto borderMetrics = props.resolveBorderMetrics(component.layoutMetrics);
+    auto rectStrokeWidth = borderMetrics.borderWidths.left;
 
     /* Check if this component has parent Paragraph component */
     RSkComponentParagraph * parent = getParentParagraph();
     ParagraphStyle paraStyle;
     std::vector<LineMetrics> metrics;
     SkScalar yOffset = 0;
     int numberOfLines = 0;
+    int maxNumberOfLines = 0;
   
     /* If parent, this text component is part of nested text(aka fragment attachment)*/
     /*    - use parent paragraph builder to add text & push style */
     /*    - draw the paragraph, when we reach the last fragment attachment*/
     if(parent) {        
-        parent->expectedAttachmentCount += data.layoutManager->buildParagraph(data.attributedString,
+        parent->expectedAttachmentCount += data.layoutManager->buildParagraph(props.backgroundColor, data.attributedString,
                                                             props.paragraphAttributes,
                                                             true,
                                                             parent->paraBuilder);
         auto paragraph = parent->paraBuilder->Build();
         parent->currentAttachmentCount++;
         if(!parent->expectedAttachmentCount || (parent->expectedAttachmentCount == parent->currentAttachmentCount)) {
             auto frame = parent->getAbsoluteFrame();
-            paragraph->layout(frame.size.width);
+            paragraph->layout(frame.size.width - (2*rectStrokeWidth));
```

Comment:
Please check if you can make use of LayoutMetrics contentInsets for getting this value

Reply:
component.layoutMetrics.getContentFrame() will give provide width & height reducing contentInsets

Reply:
Updated